Samplesheet_converter_v1.0.py

```diff
@@ -0,0 +1,108 @@
+# These scripts are for the purpose of converting Illumina samplesheet that contains Chromium 10X indexes for demultiplexing.
+# Headers and lines with ordinary indexes will be passed without any change. Lines with Chromium 10X indexes will be expanded into 4 lines, with 1 index in each line, and suffix 'Sx' will be added at the end of sample names.
+# Example:
+# ------------------------------------------------------------------------------------------------------
+# Script: python main.py -i <inputfile> -o <outputfile> -x <indexlibrary>
+# ------------------------------------------------------------------------------------------------------
+# Original samplesheet:
+# [Header]
+# Date,None
+# Investigator Name,Chuan Wang
+# Experiment Name,Project_001
+# [Reads]
+# 151
+# 151
+# [Data]
+# Lane,SampleID,SampleName,SamplePlate,SampleWell,index,index2,Project,Description
+# 1,Sample_101,101,HGWT5ALXX,1:1,SI-GA-A1,,Project_001,
+# ------------------------------------------------------------------------------------------------------
+# Modified samplesheet:
+# [Header]
+# Date,None
+# Investigator Name,Chuan Wang
+# Experiment Name,Project_001
+# [Reads]
+# 151
+# 151
+# [Data]
+# Lane,SampleID,SampleName,SamplePlate,SampleWell,index,index2,Project,Description
+# 1,Sample_101_S1,101_S1,HGWT5ALXX,1:1,GGTTTACT,,Project_001,
+# 1,Sample_101_S2,101_S2,HGWT5ALXX,1:1,CTAAACGG,,Project_001,
+# 1,Sample_101_S3,101_S3,HGWT5ALXX,1:1,TCGGCGTC,,Project_001,
+# 1,Sample_101_S4,101_S4,HGWT5ALXX,1:1,AACCGTAA,,Project_001,
+# ------------------------------------------------------------------------------------------------------
+
+# Samplesheet_convert_v1.0
+# Written by Chuan Wang (chuan-wang@github), 2017-03-06
+
+#!/usr/bin/python
+
+import sys, getopt
+import csv
+
+# Read index library
+def read_index_library(indexlibrary):
+    with open(indexlibrary,mode='r') as input_file:
+        indexes = csv.reader(input_file)
+        index_library = {rows[0]:[rows[1],rows[2],rows[3],rows[4]] for rows in indexes}
+    return index_library
+
+# Modify samplesheet
+def modify_samplesheet(inputfile,indexlibrary):
+    index_library=read_index_library(indexlibrary)
+    modified_samplesheet=[]
+    with open(inputfile,mode='r') as org:
+        samplesheet = csv.reader(org)
+        for row in samplesheet:
+            if len(row) != 9:
```

Comment:
this is a bit too restricted check. there can be situation in which a row as few or less entries. And we can have one of the first rows with 9 commas....
Use the `[Data]` field to know that you are in that frame of the file, otherwise use this library https://github.com/SciLifeLab/flowcell_parser/ that has a samplesheet parser that can be extremely useful in this case